src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -122,59 +131,69 @@ public Builder(String name, String spaceType, int m, int efConstruction) {
 
         @Override
         public KNNVectorFieldMapper build(BuilderContext context) {
-            return new KNNVectorFieldMapper(name, new KNNVectorFieldType(buildFullName(context), meta.getValue(),
-                    dimension.getValue()), multiFieldsBuilder.build(this, context),
-                    ignoreMalformed(context), spaceType, m, efConstruction, copyTo.build(), this);
-        }
-    }
+            if (spaceType == null) {
+                spaceType = getSpaceType(context.indexSettings());
+            }
```

Comment:
Why are we storing state in Builder class? Why not get a fresh copy every time by reading from indexSettings?

Reply:
Makes sense. Updated.